bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/handler/EtherRainHandler.java

```diff
@@ -0,0 +1,211 @@
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.etherrain.internal.EtherRainBindingConstants;
+import org.openhab.binding.etherrain.internal.EtherRainException;
+import org.openhab.binding.etherrain.internal.api.EtherRainCommunication;
+import org.openhab.binding.etherrain.internal.api.EtherRainStatusResponse;
+import org.openhab.binding.etherrain.internal.config.EtherRainConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EtherRainHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Joe Inkenbrandt - Initial contribution
+ */
+@NonNullByDefault
+public class EtherRainHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EtherRainHandler.class);
+
+    private @Nullable EtherRainCommunication device = null;
+    private boolean connected = false;
+    private @NonNullByDefault({}) EtherRainConfiguration config = null;
+
+    private @Nullable ScheduledFuture<?> updateJob = null;
+
+    private final HttpClient httpClient;
+
+    /*
+     * Constructor class. Only call the parent constructor
+     */
+    public EtherRainHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+        this.updateJob = null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command == RefreshType.REFRESH) {
+            scheduler.execute(this::updateBridge);
+            return;
+        } else if (channelUID.getId().equals(EtherRainBindingConstants.CHANNEL_ID_EXECUTE)) {
+            execute();
+            updateState(EtherRainBindingConstants.CHANNEL_ID_EXECUTE, OnOffType.OFF);
+        } else if (channelUID.getId().equals(EtherRainBindingConstants.CHANNEL_ID_CLEAR)) {
+            clear();
+            updateState(EtherRainBindingConstants.CHANNEL_ID_CLEAR, OnOffType.OFF);
+        }
+    }
+
+    private boolean connectBridge() {
+        logger.debug("Attempting to connect to Etherrain with config = (Host: {}, Port: {}, Refresh: {}).", config.host,
+                config.port, config.refresh);
+
+        device = new EtherRainCommunication(config.host, config.port, config.password, httpClient);
+
+        try {
+            device.commandStatus();
+        } catch (EtherRainException | IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not create a connection to the EtherRain");
+            logger.debug("Could not open API connection to the EtherRain device. Exception received: {}",
+                    e.getMessage());
+            device = null;
+            updateStatus(ThingStatus.OFFLINE);
+            return false;
+        }
```

Comment:
To deal with the null warnings you need to save a reference of your field to a local variable.
```suggestion
        EtherRainCommunication device = new EtherRainCommunication(config.host, config.port, config.password, httpClient);

        try {
            device.commandStatus();
        } catch (EtherRainException | IOException e) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
                    "Could not create a connection to the EtherRain");
            logger.debug("Could not open API connection to the EtherRain device. Exception received: {}",
                    e.getMessage());
            this.device = null;
            updateStatus(ThingStatus.OFFLINE);
            return false;
        }
        this.device = device;
```